From d454e581660f6c38a7734fb051ae134cacff334b Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Mon, 11 Dec 2023 00:06:35 +0200
Subject: [PATCH] [SW-0] wait for prstn in ppboot

as a temp WA for access issue from PIF to PSOC, ppboot waits for
prstn instead of pif, before running pif

Change-Id: Iae3a699e1718a2731912860adcbdb0dd4d6ebca1
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index ca1cf2d..7ddf4bd 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -819,6 +819,13 @@
 	u32 run_req_reg = data->cpu_cfg_base + PIF_RUN_HALT_REQ_OFF;
 	u64 start;
 
+	/*
+	 * TODO: temporary WA, this polling for PRSTN should be in PIF ARC
+	 * Wait for PRSTn
+	 */
+	while (!regmap_readl(D0_PSOC_GLOBAL_CONF_REG(mmGLOBAL_CONF_PRSTN)))
+		;
+
 	/* Set ARC in running state */
 	regmap_writel(PIF_ARC_RUN_HALT_REQ_RUN_REQ_M, run_req_reg);
 
@@ -1072,10 +1079,6 @@
 {
 	u32 link_timeout;
 	enum pif_reset_source rst_src;
-
-	/* Wait for PRSTn */
-	while (!regmap_readl(D0_PSOC_GLOBAL_CONF_REG(mmGLOBAL_CONF_PRSTN)))
-		;
 
 	/* arm timer for delays and timeouts in the pcie init sequence */
 	pif_ts_counter_arm();
